replica/tablets.cc

```diff
@@ -556,11 +559,11 @@ class tablet_incremental_selector : public sstables::incremental_selector_impl {
         if (!_cur_set || pos.token() >= _lowest_next_token) {
             auto idx = _tset.group_of(token);
             if (!token.is_maximum()) {
-                _cur_set = _tset._sstable_sets[idx];
+                _cur_set = _tset._sstable_sets.at(idx);
```

Comment:
this creates a regression.  we don't require the set to exist.  there's code a few lines below to handle `if (!_cur_set)` and return an empty list, plus a hint about the lowest_next_position.

Reply:
why? we won't go through inexistent sets anymore.

Reply:
this was only needed when you emplaced non-engaged sets.

Reply:
But select can be called with any token

Reply:
with tokens that are owned by existing tablets. what can happen is that we won't find the sstable_set, and that will be handled with your proposed find_sstable_set that aborts if id is not found, but a sstable set at an id X won't be empty as before with the sparse array. we don't have a sparse array anymore.

Reply:
select can be called with any token, and storage_proxy will consult effective_replication_map to find the node that owns the token, and fencing will prevent an oudated coordinator from reaching this area.